replica/database.cc

```diff
@@ -1411,6 +1411,9 @@ const column_family& database::find_column_family(const schema_ptr& schema) cons
 }
 
 void database::validate_keyspace_update(keyspace_metadata& ksm) {
+    if (ksm.strategy_name()  == "org.apache.cassandra.locator.NetworkTopologyStrategy" && ksm.strategy_options().empty()) {
+        throw exceptions::configuration_exception("Configuration for at least one datacenter must be present");
+    }
```

Comment:
Would be cleaner if the validation happened inside the strategy class.

Reply:
@avikivity Sure it would but cassandra_tests/validation/operations/create_test.py::testCreateKeyspaceWithNetworkTopologyStrategyNoOptions is failed.
I have no idea why cassandra allows to create a keyspace with empty options and disallows to alter it. But if we want to hide this validation in the strategy class we need to add validation for update method to abstract strategy class and to whole chain from this point. Do we really need it?

Reply:
@nyh Done